src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -270,6 +272,8 @@ internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList
                     }
```

Comment:
This won't work. `baseInterfaces` comes from the outer loop. Perhaps the suggestion in the comment should be considered?